### PR TITLE
Cleanup after init (removing log files).

### PR DESCRIPTION
**Pull request by tpfau**
Cleaning up after `initCobraToolbox` and `changeCobraSolver`.
Since changeCobraSolver now explicitly solves a trivial problem to see if the respective solver works properly, it also creates log files, which should be cleaned up after the call.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: initCobraToolbox.m [hunk 1]
```diff
@@ -95,13 +95,19 @@ function initCobraToolbox()
...
+            
     % add the external install folder
     addpath(genpath([CBTDIR filesep 'external' filesep 'install']));
 
     % add the install folder
     addpath(genpath([CBTDIR filesep 'src' filesep 'base' filesep 'install']));
 
+    %create the Cleanup function    
+    %get the current content of the init Folder
+    CobraContent = dir(CBTDIR);
```
Comment (laurentheirendt) [inline initCobraToolbox.m]: May you rename this variable here as `dirContent`?

File: initCobraToolbox.m [hunk 2]
```diff
@@ -95,13 +95,19 @@ function initCobraToolbox()
...
+    finishing = onCleanup(@() removeNewLogsFromDir(CBTDIR,CobraContent));
```
Comment (laurentheirendt) [inline initCobraToolbox.m]: This variable name is not very explicit.

**Comment by tpfau** [comment]
The PR now also cleans up after the test suite is run (removing generated files).

**Comment by tpfau** [comment]
close/reopen to retrigger CI (as reconMap test seems to be a web-issue)

**Comment by tpfau** [comment]
Anything else to do here?

File: src/base/install/removeTempCOBRAFilesFromFolder.m [hunk 3]
```diff
@@ -0,0 +1,37 @@
+function removeTempCOBRAFilesFromFolder(folder, oldcontent)
```
Comment (laurentheirendt) [inline src/base/install/removeTempCOBRAFilesFromFolder.m]: May we rename this function to `removeTmpIgnoredFiles()` or similar?

File: src/base/install/removeTempCOBRAFilesFromFolder.m [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+% Removes all log files from the cobra base folder that were not present
```
Comment (laurentheirendt) [inline src/base/install/removeTempCOBRAFilesFromFolder.m]: This description is not very explicit. May you elaborate?

File: src/base/install/removeTempCOBRAFilesFromFolder.m [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+% in the content
+%
+% USAGE:
+%     removeNewLogsFromDir(folder, content)
+%
+% INPUT:
+%     content:   a directory structure as obtained by dir
+%
+
+orig = cd(folder);
```
Comment (laurentheirendt) [inline src/base/install/removeTempCOBRAFilesFromFolder.m]: May you rename `orig` to `currentDir`?

File: src/base/install/removeTempCOBRAFilesFromFolder.m [hunk 6]
```diff
@@ -0,0 +1,37 @@
+function removeTempCOBRAFilesFromFolder(folder, oldcontent)
+% Removes all log files from the cobra base folder that were not present
+% in the content
+%
+% USAGE:
+%     removeNewLogsFromDir(folder, content)
```
Comment (laurentheirendt) [inline src/base/install/removeTempCOBRAFilesFromFolder.m]: This must be updated.

File: test/testAll.m [hunk 7]
```diff
@@ -42,6 +43,13 @@
 % run the official initialisation script
 initCobraToolbox;
 
+%Init the cleanup:
+orig = cd('test');
```
Comment (laurentheirendt) [inline test/testAll.m]: May you also rename `orig` here?

File: src/base/install/removeTempCOBRAFilesFromFolder.m (hunk 3 above)
Comment (laurentheirendt) [reply src/base/install/removeTempCOBRAFilesFromFolder.m]: May you also rename `folder` to `directory`?

File: test/testAll.m [hunk 8]
```diff
@@ -65,29 +73,15 @@
...
-        end
-    end
-
-    % close the .gitignore file
-    fclose(fid);
-
+    % Get the ignored Files from gitIgnore
+    % only retain the lines that end with .txt and .m and 
+    %are not comments and point to files in the /src folder
+    ignoredPatterns = {'^.{0,3}$',... % Is smaller than four.
```
Comment (laurentheirendt) [inline test/testAll.m]: Nice 👍

**Comment by tpfau** [comment]
I added a further small change to speed this up again, as the rdir here takes awfully long. 
As soon as initCOBRAToolbox is done, we don't need to validate solvers any more and can just return false on anything that is not marked as installed. Anything else should work and thus doesn't need the validation any more neither. 
This allows us to:
1. not run a minimal sample on every changeCobraSolver call.
2. Avoid the rdir if not in initCobraToolbox (or explicitly requesting validation).

**Comment by tpfau** [comment]
Closing for some more testing

**Comment by laurentheirendt** [comment]
Closing & re-opening to trigger CI.

**Pull request merged**